src/abi.md

```diff
@@ -57,20 +245,97 @@ $ nm -C foo.o
 0000000000000000 T foo::quux
 ```
 
-## The `no_mangle` attribute
+## Symbol naming
+
+r[abi.symbol-name]
+
+```abnf
+MetaItemNoMangle := "no_mangle"
+MetaItemExportName := "export_name" "=" ([STRING_LITERAL] | [RAW_STRING_LITERAL])
+```
+
+r[abi.symbol-name.names]
+The *`no_mangle` attribute* and the *`export_name` attribute* shall only be applied to a `static` or `fn` item. The *`export_name` attribute* shall not be applied to an item declared within an [`extern` block].
+
+```rust
+#[no_mangle]
+extern "C" fn foo(x: i32) -> i32 {
+    x + 1
+}
+
+#[export_name = "bar"]
+extern "C" fn baz(x: i32) -> i32 {
+    x + 2
+}
+```
+
+```rust,compile_fail
+extern "C" {
+    #[export_name = "foo"]
+    fn __foo(x: i32) -> i32;
```

Comment:
It can be helpful for error examples to highlight where the error is.

```suggestion
    fn __foo(x: i32) -> i32;  // error: not a free function, impl method, or static
```